script-engine/src/test/java/org/renjin/script/RenjinScriptEngineTest.java

```diff
@@ -136,5 +138,32 @@ public void readResource() throws ScriptException {
     assertThat(vector.length(), equalTo(1));
     assertThat(vector.getElementAsString(0), equalTo("hello world"));
   }
-  
+
+
+  @Test
+  public void sameBindingsFromEngineAndContext() throws ScriptException {
+
+    ScriptEngineManager scriptEngineManager = new ScriptEngineManager();
+    ScriptEngine scriptEngine = scriptEngineManager.getEngineByName("Renjin");
+    if (scriptEngine == null) {
+      throw new IllegalStateException("Could not load ScriptEngine: Renjin");
+    }
+
+    Bindings bindingsFromEngine = scriptEngine.getBindings(ScriptContext.ENGINE_SCOPE);
+
+    Bindings bindingsFromContext = scriptEngine.getContext().getBindings(ScriptContext.ENGINE_SCOPE);
+
+    assertTrue(bindingsFromEngine == bindingsFromContext); // identity test
+
+    final String key = "foo";
+    final String value = "bar";
+
+    Assert.assertNull(bindingsFromEngine.get(key));
+    Assert.assertNull(bindingsFromContext.get(key));
+
+    bindingsFromEngine.put(key, value);
+    Assert.assertEquals(bindingsFromEngine.get(key), value);
```

Comment:
@peter-gergely-horvath does the API require that value from `get(key)` is equal to the original value set? 

Renjin's implementation automatically converts scalars like instances of `java.lang.String` or `java.lang.Number' to an R object  (org.renjin.sexp.SEXP), so what you put in does not always equal what you get out.

Reply:
According to the Javadoc:
> Retrieves a value set in the state of this engine. The value might be one which was set using setValue or some other value in the state of the ScriptEngine, depending on the implementation.

https://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngine.html#get-java.lang.String-

This makes sense if you consider as example an implementation for the JavaScript language: while you put a string value for a key in your bindings, the execution of some script might result in the variable to get an array value.

As a consequence, the assertions on lines 165 and 167 have to be rewritten considering the fact that Renjin wraps the given `java.lang.String` value into an expression of type `StringArrayVector`.

```
    Assert.assertEquals(StringVector.valueOf(value), bindingsFromEngine.get(key);

    Assert.assertEquals(StringVector.valueOf(value), bindingsFromContext.get(key);
```